utils/url.tsx

```diff
@@ -185,3 +185,8 @@ export function isPermalinkURL(url: string): boolean {
 
     return isInternalURL(url, siteURL) && (regexp.test(url));
 }
+
+export function isStringContainingUrl(text: string): boolean {
```

Comment:
I have limited knowledge on Regex. I am open to any suggestion to improve this function

Reply:
As far as I can tell, that regex looks okay (I've written more than a few regexes that have caused the whole app to freeze on bad input), but I think we can probably simplify this. We only care that the string contains a URL and not what exactly that URL is, which would be a lot harder, so I think we just need the first part of yours that checks whether or not a string starts with a URL protocol. We should also add something to check if the string starts with `www.` since we also autolink links like that

Here's a couple examples of the bare minimum that we autolink currently:
![Screen Shot 2022-02-22 at 1 02 33 PM](https://user-images.githubusercontent.com/3277310/155192020-8b373af9-a41b-4563-a0da-a990cb29435e.png)

Reply:
Regex is fairly simplified, as suggested checking the following cases, url includes one of them
- `https://`
- `http://`
- `www.`

Reply:
This should be fine as long as we don't care about other protocols. From a security standpoint this is actually probably better to not generate links for other protocols :P

We could potentially want to use `mailto:` in the announcement banner, but 0/5 on whether we allow that or not.

Reply:
Strictly speaking, we support any number of different URL protocols including custom ones both with and without the `//`, but I don't expect anyone to reasonably use them here, so let's ignore those for now to avoid complicating things 😄